src/Illuminate/View/Factory.php

```diff
@@ -693,6 +700,63 @@ public function doneRendering()
     }
 
     /**
+     * Add new loop to the stack.
+     *
+     * @param  int  $length
+     * @return void
+     */
+    public function addLoop($length)
+    {
+        $this->loopsStack[] = [
+            'index' => 0,
+            'reverseIndex' => $length + 1,
+            'size' => $length,
+            'isFirst' => true,
+            'isLast' => $length == 1,
+            'depth' => count($this->loopsStack) + 1,
+            'parent' => count($this->loopsStack) ? Arr::last($this->loopsStack) : null,
+        ];
+    }
+
+    /**
+     * Pop the a loop from the top of the stack.
+     *
+     * @return void
+     */
+    public function popLoop()
+    {
+        array_pop($this->loopsStack);
+    }
+
+    /**
+     * Increment the top loop's indices.
+     *
+     * @return void
+     */
+    public function incrementLoopIndices()
+    {
+        $loop = &$this->loopsStack[count($this->loopsStack) - 1];
+
+        $loop['index'] += 1;
+
+        $loop['reverseIndex'] -= 1;
+
+        $loop['isFirst'] = $loop['index'] == 1;
+
+        $loop['isLast'] = $loop['index'] == $loop['size'];
```

Comment:
Only do this if `size` is set. (Which it should only be if the item being looped over is either an array or a object that is `Countable`)